src/game/route.py

```diff
@@ -30,8 +30,10 @@ def generate_entry_code() -> str:
 
 def get_optional_cents_form_param(name: str) -> Optional[int]:
     try:
-        amount = float(request.form.get(name))
-        return math.ceil(amount * 100)
+        tokens = request.form.get(name).split('.')
+        amount = int(tokens[0]) * 100
+        if len(tokens) > 1: amount += int(tokens[1])
```

Comment:
This seems to return `None` for inputs like `100.` and `.12`

But, we can add `or 0` to convert the falsey empty string after the dot to 0 cents with
```suggestion
        if len(tokens) > 1: amount += int(tokens[1] or 0)
```


Also turns out there's [`str.partition`](https://docs.python.org/3/library/stdtypes.html#str.partition) so we can get the parts before the dot and after, and it handles the part after the dot being conditionally there for us. We can use something like
```python
dollars, _, cents = input.partition('.')
return int(dollars or 0) * 100 + int(cents or 0)
```

Reply:
Not sure if we also want to do things like truncating the cents - this is probably good enough though

Reply:
updated